### DISCUSSION TITLE
Trouble with first attempt to build 01_wait_forever

### DISCUSSION THREAD

**Discussion by humbleaudio** [rust-embedded/rust-raspberrypi-OS-tutorials#90] [Tutorial Questions] [answered]
Hi  folks,

I'm just getting started with the tutorial, and running into a snag. I'm on Mac OS 11, and I think I got all the steps from the "tl;dr Version" (AKA setup your environment) done properly but I may be wrong there. Now, I'm running into this error. Google isn't turning up any apparently way forward. Appreciate any help!

```
wedge 01_wait_forever (master) $ make
RUSTFLAGS="-C link-arg=-Tsrc/bsp/raspberrypi/link.ld -C target-cpu=cortex-a53 -D warnings -D missing_docs" cargo rustc --target=aarch64-unknown-none-softfloat --features bsp_rpi3 --release
error: invalid channel name '[toolchain]' in '/Users/<REDACTED>/rust/rust-raspberrypi-OS-tutorials/rust-toolchain'
error: caused by: invalid toolchain name: '[toolchain]'
make: *** [target/aarch64-unknown-none-softfloat/release/kernel] Error 1
wedge 01_wait_forever (master) $ 
```

**Comment by andre-richter** [comment, marked as answer]
The `rust-toolchain` file uses the recently introduced toml format. It could be that your `rustup` installation is too old.

Maybe updating rustup itself helps.

Otherwise, the nuclear option is:
`rm -rf ~/.rustup ~/.cargo` and then follow the Readme for the "from scratch" case.

Let me know if that helps.

**Reply by humbleaudio** [reply]
Ahha! Yes, `rustup` needed an update. Thanks!
